engine/src/main/java/org/terasology/rendering/dag/WireframeTask.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import org.lwjgl.opengl.GL11;
+import static org.lwjgl.opengl.GL11.GL_FILL;
+import static org.lwjgl.opengl.GL11.GL_FRONT_AND_BACK;
+import static org.lwjgl.opengl.GL11.GL_LINE;
+
+/**
+ *
+ */
+public class WireframeTask implements PipelineTask { // TODO: separate this into SetWireframeOnTask and SetWireframeOffTask?
```

Comment:
Absolutely not. 

But thank you for asking because I now have to focus on how exactly this should work. =)

Let's remind ourselves of what I suggested in the previous PR.
- If a node desires a state change toward a default value, this isn't actually listed in the set of desired state changes - after node.process() is executed the system always brings values back to defaults, unless the next node says otherwise.
- If a node desires a state change toward a non-default value, a specialized instance implementing StateChange is in the desired set and its getValue() method returns the chosen non-default value.
- I then suggested that if node can handle any possible values of a particular OpenGL attribute an appropriate StateChange implementation is listed in the desired set, but its getValue() method should return null. 

In this scenario the RenderPipelineOptimizer would have to take, say, a WireframeStateChange instance, call its getValue() method, obtain null and... know what to do with it! I.e., for wireframe, it would need to find the current wireframe setting in the render config before it generates the WireframeStateChangeTask. That's an option but for the optimizer to know how to do this a new class would be needed, a StateChangeHandler. So, for every StateChange implementation we'd need a StateChangeHandler implementation and a StateChangeTask implementation, which sounds a bit too much. 

So, my suggestion is: StateChange.getValue() never returns null. It must always return a "good value", something that can be used as it is by the optimizer and by the StateChangeTask associated with that StateChange implentation. So, who is responsible for keeping that value the correct one for the Node? Well, I'd say the Node itself! So, if a Node wants to be rendered in wireframe depending on what the settings in the RenderConfig are, the Node needs to keep an eye on the RenderConfig, refresh the StateChange instance in its desired set when the settings change and trigger a refresh of the RenderPipeline and the OptimizedPipeline.

So, from the perspective of the RenderPipelineGenerator and the RenderPipelineOptimizer (if we wish to keep calling them this way), a StateChange instance will always have a value that can be handled as it is, and the Node implementation is responsible for providing and up to date desiredStateChanges set.